src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -160,3 +171,60 @@ void DaemonLocalServerConnection::write(const QJsonObject& obj) {
   m_socket->write(QJsonDocument(obj).toJson(QJsonDocument::Compact));
   m_socket->write("\n");
 }
+
+bool DaemonLocalServerConnection::isCallerAuthorized(const QString& command) {
+#if defined(MZ_MACOS)
+  uid_t uid;
+  gid_t gid;
+  int result = getpeereid(m_socket->socketDescriptor(), &uid, &gid);
+  if (result == -1) {
+    logger.error() << "Unable to determine if caller is authorized."
+                      "Assuming unauthorized.";
+    return false;
+  }
+
+  if (uid == 0) {
+    logger.debug() << "Command issued by root.";
+    // Case: This is root calling, root is always authorized.
+    return true;
+  }
+
+  QJsonObject status = Daemon::instance()->getStatus();
```

Comment:
There's a bit of a performance hit, and a race condition waiting to bite us here. In the case where the daemon has been requested to start a connection, but the Wireguard handshake hasn't completed yet, `Daemon::getStatus()` will return false which might result in an unexpected flow through this function.

In the case where at least one connection request has been made, then `Daemon::getStatus()` is going to make an IPC call down to the tunnel process (MacOS/Windows) or Kernel (Linux), which will add unnecessary latency to this check. It would be better to check if a connection has been requested, regardless of the status of that connection. For example, we could test if `Daemon::m_connections.isEmpty()` or by setting a bool on `activate()` and clearing it on `deactivate()`.

Reply:
Addressed in [24c36ad](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8313/commits/24c36adb3194aabc3af117c9007f2ccec0b4bfdb). LMK.